hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java

```diff
@@ -1126,7 +1126,7 @@ public HFileFilter(FileSystem fs) {
 
     @Override
     protected boolean accept(Path p, @CheckForNull Boolean isDir) {
-      if (!StoreFileInfo.isHFile(p)) {
+      if (!StoreFileInfo.isHFile(p) && !StoreFileInfo.isMobFile(p)) {
```

Comment:
if the hfile name pattern is updated to handle the newer mob hfile pattern we don't need this extra call.

Reply:
Created: https://issues.apache.org/jira/browse/HBASE-23724